policy/policy.go

```diff
@@ -64,9 +65,38 @@ func (pr *Processor) ValidatePolicy(p *sdk.ScalingPolicy) error {
 		mErr = multierror.Append(mErr, errors.New("policy Min must not be greater Max"))
 	}
 
+	if err := pr.validateHorizontalClusterPolicy(p); err != nil {
+		mErr = multierror.Append(mErr, err)
+	}
+
 	return mErr.ErrorOrNil()
 }
 
+// validateHorizontalClusterPolicy validates that the target config for a
+// horizontal cluster scaling policy is valid. Calling this function is safe in
+// situations where it is not known whether the policy is a horizontal cluster
+// target or not.
+func (pr *Processor) validateHorizontalClusterPolicy(p *sdk.ScalingPolicy) error {
+
+	// Protect against the policy not targeting a horizontal cluster target so
+	// callers don't have to perform this.
+	if !p.Target.IsNodePoolTarget() {
+		return nil
+	}
```

Comment:
It would be nice to be able to detect that there is no node class or datacenter defined in the policy, but I see how that's not possible right now.

In the future, `cluster-scaling` could be one of the capabilities that target plugins could advertise (related issue: https://github.com/hashicorp/nomad-autoscaler/issues/444).